homeassistant/components/rejseplanen/__init__.py

```diff
@@ -1 +1,101 @@
 """The rejseplanen component."""
+
+from __future__ import annotations
+
+import logging
+
+from dbus_fast import AuthError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.issue_registry import IssueSeverity, async_create_issue
+from homeassistant.helpers.typing import ConfigType
+
+from .const import DOMAIN
+from .coordinator import RejseplanenDataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+CONFIG_SCHEMA = cv.config_entry_only_config_schema(DOMAIN)
+
+type RejseplanenConfigEntry = ConfigEntry[RejseplanenDataUpdateCoordinator]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: RejseplanenConfigEntry,
+) -> bool:
+    """Set up Rejseplanen from a config entry."""
+    _LOGGER.debug(
+        "Setting up Rejseplanen integration for entry: %s", config_entry.entry_id
+    )
+    coordinator: RejseplanenDataUpdateCoordinator | None = getattr(
+        config_entry, "runtime_data", None
+    )
+    if coordinator is None:
+        coordinator = RejseplanenDataUpdateCoordinator(
+            hass,
+            config_entry,
+        )
+        config_entry.runtime_data = coordinator
+        # Test the connection/setup BEFORE forwarding to platforms
+        try:
+            await coordinator.async_config_entry_first_refresh()
+        except (ConnectionError, TimeoutError, AuthError) as err:
+            raise ConfigEntryNotReady(
+                f"Unable to connect to Rejseplanen: {err}"
+            ) from err
```

Comment:
These exceptions should be caught in the coordinator `_async_update_data` and should be re-raised as `UpdateFailed`